consent/draft-ietf-rtcweb-stun-consent-freshness.xml

```diff
@@ -334,14 +334,16 @@
         an off-path attacker spoofing TCP segments cannot cause a TCP sender
         to send once the consent timer expires (30 seconds).</t>
 
-        <t>An endpoint that is not sending any application data does not need
-        to maintain consent. However, not sending any traffic could cause NAT
-        or firewall mappings to expire. Furthermore, having one peer unable to
-        send is detrimental to many protocols. Absent better information about
-        the network, if an endpoint needs to ensure its NAT or firewall
-        mappings do not expire, it can be done using keepalive or other
-        techniques (see Section 10 of <xref target="RFC5245"></xref> and see
-        <xref target="RFC6263"></xref>).</t>
+        <t>An endpoint that is not sending any application data after
+        obtaining consent does not need to maintain consent, but the endpoint
+        MUST regain consent before it resumes sending application data.
+        However, not sending any traffic could cause NAT or firewall mappings
+        to expire. Furthermore, having one peer unable to send is detrimental
+        to many protocols. Absent better information about the network, if an
+        endpoint needs to ensure its NAT or firewall mappings do not expire,
+        it can be done using keepalive or other techniques (see Section 10 of
+        <xref target="RFC5245"></xref> and see <xref
+        target="RFC6263"></xref>).</t>
```

Comment:
Fixed.

-Tiru

From: Martin Thomson [mailto:notifications@github.com]
Sent: Monday, June 22, 2015 9:32 PM
To: tireddy/ConsentFreshness
Cc: Tirumaleswar Reddy (tireddy)
Subject: Re: [ConsentFreshness] Comments from David (#1)


In consent/draft-ietf-rtcweb-stun-consent-freshness.xml<https://github.com/tireddy/ConsentFreshness/pull/1#discussion_r32950863>:

> -        or firewall mappings to expire. Furthermore, having one peer unable to

> -        send is detrimental to many protocols. Absent better information about

> -        the network, if an endpoint needs to ensure its NAT or firewall

> -        mappings do not expire, it can be done using keepalive or other

> -        techniques (see Section 10 of <xref target="RFC5245"></xref> and see

> -        <xref target="RFC6263"></xref>).</t>

> +        <t>An endpoint that is not sending any application data after

> +        obtaining consent does not need to maintain consent, but the endpoint

> +        MUST regain consent before it resumes sending application data.

> +        However, not sending any traffic could cause NAT or firewall mappings

> +        to expire. Furthermore, having one peer unable to send is detrimental

> +        to many protocols. Absent better information about the network, if an

> +        endpoint needs to ensure its NAT or firewall mappings do not expire,

> +        it can be done using keepalive or other techniques (see Section 10 of

> +        <xref target="RFC5245"></xref> and see <xref

> +        target="RFC6263"></xref>).</t>

An endpoint does not need to maintain consent if it does not send application data. However, an endpoint MUST regain consent before it resumes sending application data. In the absence of any packets, any bindings in middleboxes for the flow might expire. Furthermore, having one peer unable to send is detrimental to many protocols.

I don't think that I would mention keepalive here. But have no strong objection to the language being there either.

—
Reply to this email directly or view it on GitHub<https://github.com/tireddy/ConsentFreshness/pull/1/files#r32950863>.